das/simple_das_reader_aggregator.go

```diff
@@ -74,6 +77,40 @@ func SimpleExploreExploitStrategyConfigAddOptions(prefix string, f *flag.FlagSet
 	f.Int(prefix+".exploit-iterations", DefaultSimpleExploreExploitStrategyConfig.ExploitIterations, "number of consecutive GetByHash calls to the aggregator where each call will cause it to select from REST endpoints in order of best latency and success rate, before switching to explore mode")
 }
 
+func addAggregatorReadersOnRestfulServerListUpdate(ctx context.Context, config *RestfulClientAggregatorConfig, a *SimpleDASReaderAggregator, combinedUrls map[string]bool) {
+	onlineUrlsChan := StartRestfulServerListFetchDaemon(ctx, config.OnlineUrlList, config.OnlineUrlListFetchInterval)
+
+	addRestfulDasClients := func(urls []string) bool {
+		for _, url := range urls {
+			if combinedUrls[url] {
+				continue
+			}
+			combinedUrls[url] = true
+			reader, err := NewRestfulDasClientFromURL(url)
+			if err != nil {
+				return false
+			}
+			a.readers = append(a.readers, reader)
+			a.stats[reader] = make([]readerStat, 0, config.MaxPerEndpointStats)
```

Comment:
I think this isn't thread safe, there's a comment in SimpleDASReaderAggregator

```
// readers and stats are only to be updated by the stats goroutine
```

Reply:
moved inside stats goroutine